Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Math/VectorN.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Vector4.h"
+
+namespace ScriptCanvas
+{
+    namespace VectorNFunctions
+    {
+        Data::VectorNType Zero(Data::NumberType size)
+        {
+            return Data::VectorNType::CreateZero(aznumeric_cast<AZStd::size_t>(size));
+        }
+
+        Data::VectorNType One(Data::NumberType size)
+        {
+            return Data::VectorNType::CreateOne(aznumeric_cast<AZStd::size_t>(size));
+        }
+
+        Data::VectorNType Random(Data::NumberType size)
+        {
+            return Data::VectorNType::CreateRandom(aznumeric_cast<AZStd::size_t>(size));
+        }
+
+        Data::NumberType GetDimensionality(const Data::VectorNType& source)
+        {
+            return static_cast<Data::NumberType>(source.GetDimensionality());
+        }
+
+        Data::NumberType GetElement(const Data::VectorNType& source, const Data::NumberType index)
+        {
+            return source.GetElement(AZ::GetClamp<AZStd::size_t>(aznumeric_cast<AZStd::size_t>(index), 0, source.GetDimensionality() - 1));
+        }
+
+        Data::BooleanType IsClose(const Data::VectorNType& a, const Data::VectorNType& b, Data::NumberType tolerance)
+        {
+            return a.IsClose(b, aznumeric_cast<float>(tolerance));
+        }
+
+        Data::BooleanType IsZero(const Data::VectorNType& source, Data::NumberType tolerance)
+        {
+            return source.IsZero(aznumeric_cast<float>(tolerance));
+        }
+
+        Data::VectorNType& ReLU(Data::VectorNType& source)
```

Comment:
So this performs a [Rectified Linear Unit](https://www.mathworks.com/help/deeplearning/ref/nnet.cnn.layer.relulayer.html) operation.
This seems machine learning specific and not part of a normal math vector API

Reply:
Yes, it's ML specific. I could remove it, but I anticipate it will be heavily used.